Add a couple small reliability refactors

pixilcode opened the pull request:
These are a couple more refactorings that I did last week while waiting for feedback. Nothing big, just random things that I noticed.

patrickwalton commented:
Looks good, just a couple thoughts:
1. Can you make sure the unused code is unused in the main branch? If not, then it could point to something that got lost in your changes.
2. Can you set up regression testing on radar.on 0.5.0 and scatterometer.on 1.0.0. You probably should pick a specific commit to test against in the process.

One other side thing that I didn't catch from previous changes until later. It looks like the reload command stopped working on your branch. You can test by running a file, leaving the repl open, and changing the file and typing reload in the repl.

pixilcode commented:
> 1. Can you make sure the unused code is unused in the main branch? If not, then it could point to something that got lost in your changes.

Yep! The `logging` and `traceback` code that I removed was used in `PythonError` to log the Python error, but that is now being handled in a uniform way in the handler/loader. I've attempted to trigger the code in the main branch by raising an error in an imported python function. However, it always gave me a name error and crashed instead.

As for the unused `parent_model` parameter, that was being passed into errors so that the errors could recursively call the loader/interpreter. However, because errors are being raised, this is no longer necessary, so I removed the parameters. I did forget to remove them from other calls to `parse_file` though, so I included a commit to correct that.

> 2. Can you set up regression testing on radar.on 0.5.0 and scatterometer.on 1.0.0. You probably should pick a specific commit to test against in the process.

Done! I've written some scripts that take "snapshots" of the output of oneil and compare against a benchmark snapshot. See [this commit](https://github.com/careweather/oneil/pull/6/commits/5ea3645448aec46795721fb059498a7a187c63da).

> One other side thing that I didn't catch from previous changes until later. It looks like the reload command stopped working on your branch.

Awesome, I fixed this! See [this commit](https://github.com/careweather/oneil/pull/6/commits/bfd528fc9f73dc18402b3c98b8d9d81b27b0ce25).

patrickwalton commented:
reload fix looks good. Learned how to add comments inline so see those.

pixilcode commented:
I don't see any comments inline in the code. Did you mean inline somewhere else?

patrickwalton commented:
Sorry, I should have been more clear. In GitHub's pull request page, you can see comments tied to lines of the diff on the commit. These show up in the discussion tab and the files changed tab.

Take care,

Patrick Walton
Co-founder/CEO
Care Weather
801-227-4740
www.careweather.com ( http://www.careweather.com )

On Fri, Jun 06, 2025 at 8:20 AM, Brendon Bown < ***@***.*** > wrote:

> 
> *pixilcode* left a comment (careweather/ oneil#6) (
> https://github.com/careweather/oneil/pull/6#issuecomment-2949417133 )
> 
> 
> I don't see any comments inline in the code. Did you mean inline somewhere
> else?
> 
> 
> 
> —
> Reply to this email directly, view it on GitHub (
> https://github.com/careweather/oneil/pull/6#issuecomment-2949417133 ) , or
> unsubscribe (
> https://github.com/notifications/unsubscribe-auth/AC4ANHYMLTKWW2O5K6AJT2L3CGPUPAVCNFSM6AAAAAB6QEUK5GVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDSNBZGQYTOMJTGM
> ).
> You are receiving this because your review was requested. Message ID: <careweather/oneil/pull/6/c2949417133
> @ github. com>
> 
> 
>

pixilcode commented:
Okay, I've updated the regression test so that it *only* uses the output of all model values and all test results. In addition, the snapshots that need to be taken are read from a file.

patrickwalton commented:
Looks great! Merged it in.

patrickwalton commented:
Looks great, merged it in. Are we ready to merge the feat-reliability branch you think?

Take care,

Patrick Walton
Co-founder/CEO
Care Weather
801-227-4740
www.careweather.com ( http://www.careweather.com )

On Mon, Jun 16, 2025 at 12:11 PM, Brendon Bown < ***@***.*** > wrote:

> 
> *pixilcode* left a comment (careweather/ oneil#6) (
> https://github.com/careweather/oneil/pull/6#issuecomment-2977564050 )
> 
> 
> Okay, I've updated the regression test so that it only uses the output of
> all model values and all test results. In addition, the snapshots that
> need to be taken are read from a file.
> 
> 
> 
> —
> Reply to this email directly, view it on GitHub (
> https://github.com/careweather/oneil/pull/6#issuecomment-2977564050 ) , or
> unsubscribe (
> https://github.com/notifications/unsubscribe-auth/AC4ANH6B7KIVOPD7H3J3XDD3D4CF5AVCNFSM6AAAAAB6QEUK5GVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDSNZXGU3DIMBVGA
> ).
> You are receiving this because your review was requested. Message ID: <careweather/oneil/pull/6/c2977564050
> @ github. com>
> 
> 
>
